chat_with_nerf/visual_grounder/visual_grounder.py

```diff
@@ -88,12 +92,28 @@ def _taking_picture(
         CONSOLE.print("[bold green]Taking 6 images... ")
         cameras.rescale_output_resolution(rendered_resolution_scaling_factor)
         cameras = cameras.to(self.lerf_pipeline.device)
-        output_filepath_path = Path(output_filename)
-        rgb_image_dir = output_filepath_path.parent / "rgb"
-        clip_image_dir = output_filepath_path.parent / "clip"
+        output_filepath_path = Path(output_filename).parent
+        # make directory for session_data if exists
+        if not os.path.exists(output_filepath_path):
+            os.makedirs(output_filepath_path)
+
+        output_filepath_path = output_filepath_path / session_id
+        # make directory for session_id if exists
+        if not os.path.exists(output_filepath_path):
+            os.makedirs(output_filepath_path)
+
+        output_filepath_path = output_filepath_path / "images"
+        # make directory for images if exists
+        if not os.path.exists(output_filepath_path):
+            os.makedirs(output_filepath_path)
+
+        rgb_image_dir = output_filepath_path / "rgb"
+        clip_image_dir = output_filepath_path / "clip"
         rgb_image_dir.mkdir(parents=True, exist_ok=True)
         clip_image_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
I don't think you need line 96-98, 101-103 or 106-108 because the .mkdir(parents=True, exist_ok=True) here will automatically create those folders for you if they don't exist.